arbor/arbexcept.cpp

```diff
@@ -4,12 +4,33 @@
 #include <arbor/arbexcept.hpp>
 #include <arbor/common_types.hpp>
 
+#include "util/unwind.hpp"
 #include "util/strprintf.hpp"
 
 namespace arb {
 
 using arb::util::pprintf;
 
+arbor_exception::arbor_exception(const std::string& what):
+    std::runtime_error{what} {
+    auto bt = util::backtrace{};
+    auto& fs = bt.frames();
+    // remove this c'tor and that of backtrace.
+    auto len = std::min(2ul, fs.size());
+    fs.erase(fs.begin(), fs.begin() + len);
```

Comment:
I agree.
And I would also be careful to check that `len` has a sane value.

Reply:
Jupp, that sounds like a good idea. 

Sane means `>=0` here, since we can never exceed the frame count and if we make it an unsigned type, not even that must be done.